### DISCUSSION TITLE
F1 as shortkey

### DISCUSSION THREAD

**Discussion by abyrnoradan** [WinEdt-Team/WinEdt-Support#61] [1.1 General Help and Discussion]
In my old winedt5.9 i redefined F1 as $^$
Is it possible with winedt11?
 I tried 
STRING="F1"
ENABLED=1
MACRO="[Ins('$$');CMD('Char Left')]"  

but nothing happened "F1" continue to launch help.

I wanted to redefine alt-T for launching LateX. As it's writen,   i changed line 3560 in Mainmenu.ini with
MACRO="Exe('%b\Exec\TeX\LaTeX.edt');"
      SHORTCUT="16460::Alt+T"       

but here too nothing happened.  "alt-T" open the "tool" menu 

I have many many other examples

What must i do ?

Thank you

**Comment by handigit** [comment]
Did you reload your ini files after changing the items? Shortcut is Ctrl+F9 or use the button in the options dialog button bar.

Best regards
Hanno

23.08.2024 16:54:08 abyrnoradan ***@***.***>:

**Comment by abyrnoradan** [comment]
Of course !
I "rebuild all"  many times .
And this morning  F1 works !!! ... But F3 as '{}' doesn't ???? 
I close, reopen, rebuild, and nothing  It's totaly anarchic !

I redefined  2 days ago ';' as '\'   (better for french keyboard)  It work yesterday  and no more today !!

How is it possible to work if the program changes  what i did as he wants ?

Each day I get something else !

Is Winedt 11 really stable ? Or completely bugged ?

**Comment by handigit** [comment]
I tried some of your changes today. Maybe in fact the standard Windows menu shortcuts like Alt+T for the tools menu override the custom shortcuts defined by the user. I tried to define some shortcut like Alt+L for LaTeX and that worked immediately. After pressing the "Load current script" button of course. I am not sure if your "Rebuild all" provides the functionality you actually need.

And yes, WinEdt 11 is in fact a stable and wonderful piece of software. In my experience the program doesn't change anything by itself. The reason for almost any issue I had with WinEdt was somewhere between my display and my chair. And was solved by reading the documentation or asking the excellent community here. I don't think that ranting about WinEdt or considering that a software should provide the same look and feel even if the version number almost doubled since your last update would be very helpful.

Best regards
Hanno

27.08.2024 13:00:11 abyrnoradan ***@***.***>:

**Comment by Alex-Simonic** [comment]
Your definition applies to active strings:

```
STRING="F1"
ENABLED=1
MACRO="[Ins('$$');CMD('Char Left')]"
```

This does not redefine the F1 key (and it did not in WinEdt 5 either). This macro is executed if you type F followed by 1. But that is not what you want. To change the F1 shortcut you must define a menu item with an assigned shortcut. If the same shortcut is already used for some other item you should comment it out or make sure that your item is defined before the one that you don't want to use. The first item is executed if a shortcut is used more than once. I am not aware of any problems or bugs related to this functionality... OK?
